src/main/java6/net/finmath/time/businessdaycalendar/BusinessdayCalendar.java

```diff
@@ -70,86 +70,76 @@ public LocalDate getRolledDate(LocalDate baseDate, int businessDays) {
-	 * 
-	 * First we create a new date by "adding" an offset to the base date.
-	 * The offset may be given by codes like 1D, 2D, 1W, 2W, 1M, 2M, 3M,
-	 * 1Y, 2Y, etc., where the letters denote the units as follows: D denotes days, W denotes weeks, M denotes month
-	 * Y denotes years.
-	 * 
-	 * Next the result is adjusted according to the given dateRollConvention.
-	 * 
-	 * @param baseDate The start date.
-	 * @param dateOffsetCode String containing date offset codes (like 2D, 1W, 3M, etc.) or combination of them separated by spaces.
-	 * @param dateRollConvention The date roll convention to be used for the adjustment.
-	 * @return The adjusted date applying dateRollConvention to the given date.
+	/* (non-Javadoc)
+	 * @see net.finmath.time.businessdaycalendar.BusinessdayCalendarInterface#getAdjustedDate(LocalDate, String, DateRollConvention)
 	 */
 	public LocalDate getAdjustedDate(LocalDate baseDate, String dateOffsetCode, DateRollConvention dateRollConvention) {
-		return this.getAdjustedDate(createDateFromDateAndOffsetCode(baseDate, dateOffsetCode), dateRollConvention);
+		return getAdjustedDate(createDateFromDateAndOffsetCode(baseDate, dateOffsetCode), dateRollConvention);
 	}
 
-	/**
-	 * Create a new date by "adding" a year fraction to a given base date.
-	 * 
-	 * <p>
-	 * The date offset may be given by codes like 1D, 2D, 1W, 2W, 1M, 2M, 3M,
-	 * 1Y, 2Y, etc., where the letters denote the units as follows: D denotes days, W denotes weeks, M denotes month
-	 * Y denotes years. If the date offset does not carry a letter code at the end, it will
-	 * be interpreted as ACT/365 year fraction.
-	 * </p>
-	 * 
-	 * <p>
-	 * The function may be used to ease the creation of maturities in spreadsheets.
-	 * </p>
-	 * 
-	 * @param baseDate The start date.
-	 * @param dateOffsetCode String containing date offset codes (like 2D, 1W, 3M, etc.) or combination of them separated by spaces.
-	 * @return A date corresponding the date adding the offset to the start date.
+	/* (non-Javadoc)
+	 * @see net.finmath.time.businessdaycalendar.BusinessdayCalendarInterface#createDateFromDateAndOffsetCode(LocalDate, createDateFromDateAndOffsetCode)
 	 */
-	public static LocalDate createDateFromDateAndOffsetCode(LocalDate baseDate, String dateOffsetCode) {
+	public LocalDate createDateFromDateAndOffsetCode(LocalDate baseDate, String dateOffsetCode) {
 		dateOffsetCode = dateOffsetCode.trim();
 
 		StringTokenizer tokenizer = new StringTokenizer(dateOffsetCode);
 		
-		LocalDate maturity = baseDate;
+		LocalDate maturityDate = baseDate;
 		while(tokenizer.hasMoreTokens()) {
 			String maturityCodeSingle = tokenizer.nextToken();
-
-			char unitChar = maturityCodeSingle.toLowerCase().charAt(maturityCodeSingle.length()-1);
-
+			// get unit identifier (usually last char but may be last two chars)
+			char unitChar;
+			int maturityValue;
+			if(maturityCodeSingle.toLowerCase().substring(maturityCodeSingle.length()-2).equals("bd")) {
```

Comment:
I've added the enum and implemented the conversion.

Note: I have re-enabled that ACT/365 fractions can be passed. I do understand that this may cause errors (e.g. if someone forgets the D in the offset code), but the feature is still stated in the JavaDoc and I am not sure if the feature is used.